src/commands/api/request/rest.ts

```diff
@@ -56,30 +62,50 @@ export class Rest extends SfCommand<void> {
 
     const org = flags['target-org'];
     const streamFile = flags['stream-to-file'];
-    const headers = flags.header ? getHeaders(flags.header) : {};
+    const fileOptions = flags.file
+      ? (JSON.parse(fs.readFileSync(flags.file, 'utf8')) as {
+          body?: string;
+          header?: string[];
+          url?: string;
+          method?: string;
+        })
+      : {};
 
+    if (!args.url && !fileOptions.url) {
+      throw new SfError("The url is required either in --file file's content or as an argument");
+    }
+
+    const headers = getHeaders(flags.header ?? fileOptions.header ?? []);
+    // the conditional above ensures we either have an arg or it's in the file
+    const specified = args.url ?? fileOptions.url!;
     // replace first '/' to create valid URL
-    const endpoint = args.endpoint.startsWith('/') ? args.endpoint.replace('/', '') : args.endpoint;
+    const endpoint = specified.startsWith('/') ? specified.replace('/', '') : specified;
     const url = new URL(
       `${org.getField<string>(Org.Fields.INSTANCE_URL)}/services/data/v${
         flags['api-version'] ?? (await org.retrieveMaxApiVersion())
       }/${endpoint}`
     );
 
-    const body =
-      flags.method === 'GET'
-        ? undefined
-        : // if they've passed in a file name, check and read it
-        existsSync(join(process.cwd(), flags.body ?? ''))
-        ? readFileSync(join(process.cwd(), flags.body ?? ''))
-        : // otherwise it's a stdin, and we use it directly
-          flags.body;
+    // because flags.method defaults to "GET" read from file first
+    const method = fileOptions.method ?? flags.method;
+
+    let body;
+    if (method !== 'GET') {
+      if (flags.body) {
+        body = existsSync(join(process.cwd(), flags.body))
+          ? readFileSync(join(process.cwd(), flags.body))
+          : // otherwise it's a stdin, and we use it directly
+            flags.body;
+      } else {
+        body = JSON.stringify(fileOptions.body);
```

Comment:
cc @VivekMChawla 

Today `--body` be a body string, `-` to read from stdin and a file path to read the body from.
Since this PR adds support for passing a request as a file I feel we should drop `--body` from supporting file paths (it's the same feature but we are opening 2 ways to use it).

Reply:
@WillieRuemmele - I think that @cristiand391's idea to drop support for file paths in the `--body` flag makes sense.